.github/workflows/pythonpackage.yml

```diff
@@ -40,6 +40,10 @@ jobs:
         python -m detection_rules dev license-check
 
     - name: Unit tests
+      env:
+        # only run the test test_rule_change_has_updated_date on pull request events to main
+        GITHUB_EVENT_NAME: "${{ github.event_name}}"
+        GITHUB_BASE_REF: "${{ github.event.pull_request.base.ref }}"
```

Comment:
I can't tell how this will only run the test case on main vs only running the entire step (all unit tests) only on main.  We essentially need all unit tests to run on our release branches, so if it does limit unit tests to run on only the main branch, then we should rethink this.

Reply:
At this point we are not at all limiting test case running we are just pushing 2 env variables for pytest usages. 
The actual skip is on this line. https://github.com/elastic/detection-rules/pull/3818/files#diff-143495f0794a3e8b322e5cbf9a4735acab2213fd6d0f678149b78d322976507bR643

If you look at the local testing where we have simulated running on release branches you see all test cases executed and this being skipped